internal/encode_args.go

```diff
@@ -80,6 +80,11 @@ func decodeArgsToRawValues(dc converter.DataConverter, fnType reflect.Type, data
 	}
 
 	// Unmarshal
+
+	// Dynamic workflows take in a single EncodedValues, encode all data into single EncodedValues
```

Comment:
I thought it would be varargs of `RawValue`, but encoded values makes sense too

Reply:
It is unfortunate we have both concepts, but `EncodedValues ` is more appropriate for dynamic workflows

Reply:
Hrmm. I think they're similar from a user POV for params (I personally prefer codec to apply outside workflow boundary), but we should allow any return value IMO.

Reply:
Yeah the dynamic workflow should be allowed to return anything I agree, I was referring to the input type